gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/jupiter/handlers/api/SyncApiReactor.java

```diff
@@ -253,13 +284,8 @@ private Completable invokeBackend(final RequestExecutionContext ctx) {
                 }
             )
             .doOnSubscribe(disposable -> ctx.request().metrics().setApiResponseTimeMs(System.currentTimeMillis()))
-            .doOnTerminate(
-                () ->
-                    ctx
-                        .request()
-                        .metrics()
-                        .setApiResponseTimeMs(System.currentTimeMillis() - ctx.request().metrics().getApiResponseTimeMs())
-            );
+            .doOnDispose(() -> setApiResponseTimeMetric(ctx))
+            .doOnTerminate(() -> setApiResponseTimeMetric(ctx));
```

Comment:
I think you could replace both by only dofinally to cover explicit dispose, cance, complete or on Error

Reply:
same comment as above